File: src/macos/AzPin/Models/PinnedResourceGroup.swift
import SwiftData

@Model
final class PinnedResourceGroup {
    var id: String
    var subscriptionId: String
    var name: String
    var displayOrder: Int
    @Relationship(deleteRule: .cascade) var resources: [PinnedResource]

    init(id: String, subscriptionId: String, name: String, displayOrder: Int) {
        self.id = id
        self.subscriptionId = subscriptionId
        self.name = name
        self.displayOrder = displayOrder
        self.resources = []
    }
}

File: src/windows/AzPin.Windows.Core/Models/Entities/PinnedResourceGroup.cs
using Microsoft.EntityFrameworkCore;

namespace AzPin.Windows.Models.Entities;

[Index(nameof(SubscriptionId), nameof(Name), IsUnique = true)]
public class PinnedResourceGroup
{
    public int LocalId { get; set; }
    public string Id { get; set; } = string.Empty;
    public string SubscriptionId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public int DisplayOrder { get; set; }
    public ICollection<PinnedResource> Resources { get; set; } = [];
}
